pgcli/pgexecute.py

```diff
@@ -263,15 +267,14 @@ def run(self, statement, pgspecial=None, on_error=ON_ERROR_RESUME):
                 _logger.error("traceback: %r", traceback.format_exc())
 
                 if (isinstance(e, psycopg2.OperationalError)
-                        or on_error == ON_ERROR_RAISE):
+                        or not exception_formatter):
```

Comment:
Let's change this to `if (isinstance(e, psycopg2.OperationError) or or not on_error_resume):`. 

Reason being it is possible for us to provide an `exception_formatter()` but not use it because the option is set to `STOP`. A user might change this at runtime which will merely change the option to `RESUME` without having to worry about the `exception_formatter()`. 

Also using the presence of `exception_formatter` to determine whether or not to `raise` is not very explicit and it's not self documenting.